packages/grid/_modules_/grid/hooks/root/useNativeEventListener.ts

```diff
@@ -12,6 +12,15 @@ export const useNativeEventListener = (
 ) => {
   const logger = useLogger('useNativeEventListener');
   const [added, setAdded] = React.useState(false);
+  const handlerRef = React.useRef(handler);
+
+  const wrapHandler = React.useCallback((args) => {
+    return handlerRef.current && handlerRef.current(args);
+  }, []);
+
+  React.useEffect(() => {
+    handlerRef.current = handler;
+  }, [handler]);
```

Comment:
useEventCallback is badly written, and has the wrong typings. TBH I don't want to use it. It's a small piece of code that I prefer to have inside the grid, as I have full control of it.

Reply:
> useEventCallback is badly written

Then please fix it with a pull request in the core. It was written by one of the initial autors of React: https://github.com/facebook/react/issues/14099#issuecomment-440013892.

> and has the wrong typings

This is already fixed.

>  I prefer to have inside the grid, as I have full control of it.

From my perspective 1.  it should use useLayoutEffect, not useEffect 2. it should spread all the arguments, not only the first one. 3. I have seen dozens of places in the casebase where we could use it instead of useCallback. It would give us more speed without trading much.